plugins/kotlin/j2k/shared/tests/test/org/jetbrains/kotlin/j2k/AbstractJavaToKotlinConverterSingleFileTest.kt

```diff
@@ -138,26 +149,39 @@ abstract class AbstractJavaToKotlinConverterSingleFileTest : AbstractJavaToKotli
             }
         }
 
-    private fun convertJavaToKotlin(prefix: String, javaCode: String, settings: ConverterSettings): String =
+    private fun convertJavaToKotlin(prefix: String, javaCode: String, settings: ConverterSettings, directives: Directives): String =
         when (prefix) {
             "expression" -> expressionToKotlin(javaCode, settings)
             "statement" -> statementToKotlin(javaCode, settings)
             "method" -> methodToKotlin(javaCode, settings)
-            "file" -> fileToKotlin(javaCode, settings)
+            "file" -> fileToKotlin(
+                javaCode,
+                settings,
+                if (directives.contains("INCLUDE_J2K_PREPROCESSOR_EXTENSIONS")) listOf(dummyPreprocessorEP) else emptyList()
+            )
+
             else -> error("Specify what it is: method, statement or expression using the first line of test data file")
         }
 
-    open fun fileToKotlin(text: String, settings: ConverterSettings): String {
+    open fun fileToKotlin(text: String, settings: ConverterSettings, preprocessorExtensions: List<J2kPreprocessorExtension>): String {
         val file = createJavaFile(text)
         val j2kKind = if (isFirPlugin) K2 else K1_NEW
         val extension = J2kConverterExtension.extension(j2kKind)
         val converter = extension.createJavaToKotlinConverter(project, module, settings)
         val postProcessor = extension.createPostProcessor()
-        return converter.filesToKotlin(listOf(file), postProcessor).results.single()
+        var converterResult: FilesResult? = null
+        project.executeCommand("J2K") {
```

Comment:
We need this machinery for coroutines to work, right?

Reply:
Unfortunately yes